tests/runner/src/env.rs

```diff
@@ -635,3 +739,117 @@ impl Display for ResultDisplayer {
+                    "{}",
+                    ResultDisplayer {
+                        result: Ok(Output::new_with_affected_rows(*affected_rows as usize)),
+                    }
+                )?;
+                return Ok(());
+            }
+            PgRow::RowDescription(desc) => Arc::new(Schema::new(
+                desc.iter()
+                    .map(|column| {
+                        ColumnSchema::new(column.name(), ConcreteDataType::string_datatype(), false)
+                    })
+                    .collect(),
+            )),
+            _ => unreachable!(),
+        };
+        if schema.num_columns() == 0 {
+            return Ok(());
+        }
+
+        // convert to string vectors
+        let mut columns: Vec<StringVectorBuilder> = (0..schema.num_columns())
+            .map(|_| StringVectorBuilder::with_capacity(schema.num_columns()))
+            .collect();
+        for row in self.rows.iter().skip(1) {
+            if let PgRow::Row(row) = row {
+                for (i, column) in columns.iter_mut().enumerate().take(schema.num_columns()) {
+                    column.push(row.get(i));
+                }
+            }
+        }
+        let columns: Vec<VectorRef> = columns
+            .into_iter()
+            .map(|mut col| Arc::new(col.finish()) as VectorRef)
+            .collect();
+
+        // construct recordbatch
+        let recordbatches = RecordBatches::try_from_columns(schema, columns)
+            .expect("Failed to construct recordbatches from columns. Please check the schema.");
+        let result_displayer = ResultDisplayer {
+            result: Ok(Output::new_with_record_batches(recordbatches)),
+        };
+        write!(f, "{}", result_displayer)?;
+
+        Ok(())
+    }
+}
+
+struct MysqlFormatter {
+    pub rows: Vec<MySqlRow>,
+}
+
+impl Display for MysqlFormatter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.rows.is_empty() {
+            return f.write_fmt(format_args!("(Empty response)"));
+        }
+        // create schema
+        let head_column = &self.rows[0];
+        let head_binding = head_column.columns();
+        let names = head_binding
+            .iter()
+            .map(|column| column.name_str())
+            .collect::<Vec<Cow<str>>>();
+        let schema = Arc::new(Schema::new(
+            names
+                .iter()
+                .map(|name| {
+                    ColumnSchema::new(name.to_string(), ConcreteDataType::string_datatype(), false)
+                })
+                .collect(),
+        ));
+
+        // convert to string vectors
+        let mut columns: Vec<StringVectorBuilder> = (0..schema.num_columns())
+            .map(|_| StringVectorBuilder::with_capacity(schema.num_columns()))
+            .collect();
+        for row in self.rows.iter() {
+            for (i, name) in names.iter().enumerate() {
+                columns[i].push(row.get::<String, &str>(name).as_deref());
```

Comment:
Do we need to deal with other types?

Reply:
No, similar to pg's [situation](https://docs.rs/mysql/25.0.1/mysql/prelude/trait.Queryable.html#tymethod.query_iter).